pyro/contrib/epidemiology/sir.py

```diff
@@ -483,3 +476,141 @@ def predict(self, forecast=0):
         samples["first_infection"] = samples["I"].cumsum(-1).eq(0).sum(-1)
 
         return samples
+
+
+class RegionalSIRModel(CompartmentalModel):
+    """
+    Susceptible-Infected-Recovered model with coupling across regions.
+
+    To customize this model we recommend forking and editing this class.
+
+    This is a stochastic discrete-time discrete-state model with three
+    compartments: "S" for susceptible, "I" for infected, and "R" for
+    recovered individuals (the recovered individuals are implicit: ``R =
+    population - S - I``) with transitions ``S -> I -> R``.
+
+    This model demonstrates:
+
+    1.  How to create a regional model with a ``population`` vector.
+    2.  How to model both homogeneous parameters (here ``R0``) and
+        heterogeneous parameters with hierarchical structure (here ``rho``)
+        using ``self.region_plate``.
+    3.  How to approximately couple regions in :meth:`transition_bwd` using
+        ``prev["I_approx"]``.
+
+    :param torch.Tensor population: Tensor of per-region populations, defining
+        ``population = S + I + R``.
+    :param torch.Tensor coupling: Pairwise coupling matrix. Entries should be
```

Comment:
nit: can we improve this docstring? it makes it sound like entries larger than 1 are not multiplied by R0....

Reply:
I've added a larger paragraph to the main body.